video-as-img loses alpha channel

jernoble opened the pull request:
#### 0962e25391d210a563b4ec52bee8b8d6e6e32080
<pre>
video-as-img loses alpha channel
<a href="https://bugs.webkit.org/show_bug.cgi?id=243105">https://bugs.webkit.org/show_bug.cgi?id=243105</a>
&lt;rdar://96582168&gt;

Reviewed by Geoffrey Garen.

In 251557@main, a step was added where the results of a image decode were drawn to
a ShareableBitmap to facilitate transferring the results between the GPU and WebContent
processes. This bitmap was explicitly set to be &quot;opaque&quot;. However, the results of decoding
a frame of video will always generate a CGImage with valid alpha channel information, even
if that information is fully opaque, and in the case where the source video content has a
valid alpha channel, this information is lost at this drawing step.

* LayoutTests/media/content/video-with-alpha.mov: Added.
* LayoutTests/media/video-as-img-with-alpha-expected.html: Added.
* LayoutTests/media/video-as-img-with-alpha.html: Added.
* Source/WebKit/GPUProcess/media/RemoteImageDecoderAVFProxy.cpp:
(WebKit::RemoteImageDecoderAVFProxy::createFrameImageAtIndex):

Canonical link: <a href="https://commits.webkit.org/252836@main">https://commits.webkit.org/252836@main</a>
</pre>

geoffreygaren commented on Source/WebKit/GPUProcess/media/RemoteImageDecoderAVFProxy.cpp:
```diff
@@ -139,7 +139,7 @@ void RemoteImageDecoderAVFProxy::createFrameImageAtIndex(ImageDecoderIdentifier
     DestinationColorSpace colorSpace { CGImageGetColorSpace(frameImage.get()) };
 
-    auto bitmap = ShareableBitmap::create(IntSize(width, height), { WTFMove(colorSpace), true /* isOpaque */ });
+    auto bitmap = ShareableBitmap::create(IntSize(width, height), { WTFMove(colorSpace), false /* isOpaque */ });
```
I prefer the style of making a local variable "bool isOpaque = false". I think it reads better at the call site.

geoffreygaren reviewed (approved):
r=me

jernoble committed:
video-as-img loses alpha channel
https://bugs.webkit.org/show_bug.cgi?id=243105
<rdar://96582168>

Reviewed by Geoffrey Garen.

In 251557@main, a step was added where the results of a image decode were drawn to
a ShareableBitmap to facilitate transferring the results between the GPU and WebContent
processes. This bitmap was explicitly set to be "opaque". However, the results of decoding
a frame of video will always generate a CGImage with valid alpha channel information, even
if that information is fully opaque, and in the case where the source video content has a
valid alpha channel, this information is lost at this drawing step.

* LayoutTests/media/content/video-with-alpha.mov: Added.
* LayoutTests/media/video-as-img-with-alpha-expected.html: Added.
* LayoutTests/media/video-as-img-with-alpha.html: Added.
* Source/WebKit/GPUProcess/media/RemoteImageDecoderAVFProxy.cpp:
(WebKit::RemoteImageDecoderAVFProxy::createFrameImageAtIndex):

Canonical link: https://commits.webkit.org/252836@main
